tensilelite/Tensile/KernelWriterAssembly.py

```diff
@@ -1975,8 +1975,8 @@ def graWorkGroup(self, kernel):
           module.add(wgmLabelPositive)
         else:
           module.add(SAbsI32(dst=sgpr("WGM"), src=sgpr("WGM"), comment="abs(WGM)"))
-          module.add(SCmpLeI32(src0=sgpr("WGM"), src1=1, comment="WGM <= 1 ?"))
-          module.add(SCBranchSCC1(labelName=wgmLabel.getLabelName(), comment="branch if WGM <= 1"))
+          module.add(SCmpEQI32(src0=sgpr("WGM"), src1=0, comment="WGM = 0 ?"))
```

Comment:
I think this changes the behavior of WGM=1.
What you can do is 
- move this s_cmp and branch below before s_abs (line 1978-1979 above 1977)
- change the condition to >=0

Then, WGM=0,1 are same as before and only -1 is different.
We do not actually need WGM calculation for -1 (just swap WG0 and WG1 should be enough).
That is optional. At least, please make sure you did not make any change for WGM!=-1.

Reply:
Fixed